### ISSUE TITLE
Implement Load Cell Playlist

### ISSUE THREAD

**Issue by stephenrparsons** [Lexcelon/batlab-toolkit-gui#11]
The main window has a tab for the cell playlist. This is a stacked widget that can show either cell playlist not loaded (just shows buttons for opening or creating a cell playlist - already mostly implemented), or cell playlist loaded (would show the cell playlist options and allow for editing them - not implemented and what this issue is for).

https://github.com/Lexcelon/batlab-toolkit-gui/blob/5613833bab74760859d19b542277ea4e13ecd166/src/batlabmainwindow.cpp#L105-L106

A large part of this implementation will be in the below function:

https://github.com/Lexcelon/batlab-toolkit-gui/blob/5613833bab74760859d19b542277ea4e13ecd166/src/batlabmainwindow.cpp#L297-L302

Where it will first let the user select a file to open (likely using [QFileDialog::getOpenFileName](https://doc.qt.io/qt-5/qfiledialog.html#getOpenFileName)), and will then load it into a BatlabSettings object with the below:

https://github.com/Lexcelon/batlab-toolkit-gui/blob/5613833bab74760859d19b542277ea4e13ecd166/src/batlabcore/batlabsettings.cpp#L95

And will then actually load it into the GUI and display the values and buttons to edit them. This should very closely resemble the New Cell Playlist wizard Configure Playlist Settings screen (including advanced options), just all displayed in the actual program. Once tests are running, the ability to change any of these values will be greyed out. It should also have buttons to open a different playlist, create a new one (either of which would replace the values with the new ones), or save the current one (in case the user loads one and then makes changes and wants to save it again, or wants to save it to a different filename).

The page could look something like this hideous mouse drawing:

![image](https://user-images.githubusercontent.com/4584652/37237949-d801a304-23e9-11e8-911c-12dc143c0e0d.png)

This should also be called automatically when the user goes through the New Cell Playlist (at the end it should load their playlist into the interface and be ready to start executing those tests).

Note that for the values and their editors, some of them interact with each other so when one value is updated the max/min values for other fields need to change dynamically. See below for how this is handled in the wizard:

https://github.com/Lexcelon/batlab-toolkit-gui/blob/5613833bab74760859d19b542277ea4e13ecd166/src/dialogs/newcellplaylistwizard/newcellplaylistwizard.cpp#L188-L205

**Comment by hapney** [comment]
@stephenrparsons Can I get an example cell playlist csv file and corresponding cell files?

**Comment by stephenrparsons** [comment]
Cell playlist (it's actually a JSON): https://gist.github.com/stephenrparsons/4525d3f2b7d47d503674314396811774 

Working on the corresponding cell files...

**Comment by stephenrparsons** [comment]
[batlab-log_Breakin.zip](https://github.com/Lexcelon/batlab-toolkit-gui/files/1905445/batlab-log_Breakin.zip)

**Comment by hapney** [comment]
Okay, so I've spoken to Dan a bit about this. It looks like there is a Level 2 data file included in the zip file above. In that file, a couple of the cells have a summary row, but not all. The summary row only has Capacity, CapacityRange, ColoumbicEfficiency, Impedance, AvgVoltage, and AvgCurrent, which is different from the headers of the Results table I had in a previous issue.

My questions are:
1. Should I change the Results table to have those columns?
2. What do I include in the table for those cells which have no summary row yet? Do I refer to the most recent Test Type?

**Comment by stephenrparsons** [comment]
This has been tricky for me and is something I thought we had better defined than we did. From talking to Dan earlier this week he and I decided we could have the lvl1 (known as "raw" to the user) data in each cell file and then lvl2 (intermediate summary after each charge or discharge) and lvl3 (summary of entire cell run) (together known as "summary" to the user) data in a summary file. So each cell will have its own raw file with lvl1, and there will be a summary file with lvl2 data from during the runs and lvl3 data after a test is complete. I left out the lvl3 data for a cell or two in that file to simulate just that they were not yet done with their tests.

We had earlier made the Results columns based on lvl2 data, which I had not realized is created with each charge and discharge. Really, those should show the lvl3 data, aka summary data corresponding to an entire test on a cell.

Those lvl3 columns are based on a script Dan had given me that was used for some post-processing for the solar car team. @dcambron that's all I've gone off of for the final summary data for a cell test. If that is what the final format should be, we can stick with it. If it should be different, we should define that since this has been tricky for a long time to nail down.
